Algorithm.Framework/Portfolio/EqualWeightingPortfolioConstructionModel.cs

```diff
@@ -48,16 +50,24 @@ public override IEnumerable<IPortfolioTarget> CreateTargets(QCAlgorithmFramework
                 _removedSymbols = null;
             }
 
-            if (_securities.Count == 0)
+            if (insights.Length == 0)
             {
                 return Enumerable.Empty<IPortfolioTarget>();
             }
+            
+            // Get symbols that have emit insights and still in the universe
+            var symbols = _insightCollection.Select(x => x.Symbol).Distinct().ToList();
```

Comment:
will this contain any symbols from `_removedSymbols` ? I'm thinking that removed symbols shouldn't count towards the denominator of equal weighting. Also, this has approach may have unexpected consequences when considering alpha models that don't emit all insights at one time (such an an EMA cross). For example, we have a universe with 500 securities and we get our first EMA cross. With this logic, we'll give that security 100% weighting. 20 minutes later our next security can experience an EMA cross and then we'll give him 50%, and so on. This was the original reason this looks at the universe size for the denominator instead of the count of unique securities that have had insights generated.